docs/src/debugging-verification-failures.md

```diff
@@ -0,0 +1,105 @@
+# Debugging verification failures
+
+When the result of a certain check comes back as a `FAILURE`,
+Kani offers different options to help debug:
+* `--concrete-playback`. This _experimental_ feature generates a Rust unit test case that plays back a failing
+proof harness using a concrete counterexample.
+* `--visualize`. This feature generates an HTML text-based trace that
+enumerates the execution steps leading to the check failure.
+
+## Concrete playback
+
+This section describes the concrete playback feature in more detail.
+
+### Setup
+
+The Kani library needs to be linked as a dev dependency to the crate you're trying to debug.
+This requires adding the following lines to your `Cargo.toml` file,
+which differ depending on what version of the Kani library you would like to use:
+* The latest version:
+```toml
+[dev-dependencies]
```

Comment:
Maybe change this to [target.'cfg(Kani)'.dependencies]

So this only gets pulled when user sets `--cfg kani`.

Reply:
There are ways around that, but we can tune that later. Thanks